### DISCUSSION TITLE
Upstreaming MPCC controller?

### DISCUSSION THREAD

**Discussion by hidmic** [utiasDSL/safe-control-gym#115] [Ideas]
Now that the Safe Robot Learning Competition is over, we at Ekumen were wondering if you guys at UTIAS DSL would be interested in adding an MPCC controller to the set of supported controllers. We could adapt our implementation to fit the typical `safe-control-gym` controller structure and submit it.

Thoughts? FYI @JacopoPan @adamhall

**Comment by JacopoPan** [comment]
I do think that would be very nice and, by "typical" you also mean force control as in the `main` branch?
I want to know what @adamhall thinks but it might be worthwhile to work from the `dev-experiment-class` branch already.

**Reply by hidmic** [reply]
Friendly ping @adamhall

**Reply by adamhall** [reply]
Sorry! I'm just in vacation but will be back next week!

**Comment by adamhall** [comment]
Hey! I'm back from vacation. We can certainly take a look at mixing in the MPCC controller. It would probably be best to put it in the `dev-experiment-class` branch. The major difference is that the dynamics model might be pretty different now given that the inputs will be thrusts which I believe is different than what we put for the competition. I can take a look at your implementation to see what might be required. Where might I find it?

**Reply by JacopoPan** [reply]
@hidmic and the team have a copy of the repo at https://github.com/ekumenlabs/safe-control-gym created for the competition
I think now we should work through branches/PRs on this one instead

**Reply by hidmic** [reply]
That's about right @JacopoPan, and yes, we sure can make a PR to this repository, to the `dev-experiment-class` branch. 

Regarding implementation changes, the dynamics model should be roughly the same and having full state observations and direct access to actuators may actually simplify things a bit (no need for estimators). FYI @glpuga.

**Reply by hidmic** [reply]
@adamhall friendly ping

**Reply by hidmic** [reply]
@adamhall @JacopoPan friendly ping, the last one before closing this issue internally.
